torch/csrc/autograd/variable.h

```diff
@@ -256,6 +263,35 @@ struct TORCH_API AutogradMeta : public c10::AutogradMetaInterface {
   }
 };
 
+struct TORCH_API ViewInfo {
+  /// The base `Variable`
+  /// This cannot be a view of the same type as this one.
+  Variable base_;
+
+  /// By default we use as_strided to recover views which is more efficient.
+  /// view_fn is only saved when as_strided is not supported.
+  /// If view_fn has value, we use it to recover views in backward.
+  c10::optional<std::function<Variable(const Variable&)>> view_fn_;
+
+  bool has_view_fn() const {
+    return view_fn_.has_value();
+  }
+
+  std::function<Variable(const Variable&)> view_fn() const {
+    TORCH_CHECK(has_view_fn(), "Can only access the view function if it exists.");
+    return view_fn_.value();
+  }
+
+  ViewInfo chain(const Variable & base, const Variable & tensor,
```

Comment:
What does this do?

Reply:
Ho the description got lost in a rebase. My bad :/

Reply:
Ho the description got lost in a rebase. My bad :/